lib/handler/mruby/http_request.c

```diff
@@ -184,17 +184,18 @@ static void post_response(struct st_h2o_mruby_http_request_context_t *ctx, int s
 
 static void post_error(struct st_h2o_mruby_http_request_context_t *ctx, const char *errstr)
 {
-    static const h2o_header_t headers_sorted[] = {
-        {&H2O_TOKEN_CONTENT_TYPE->buf, NULL, {H2O_STRLIT("text/plain; charset=utf-8")}},
-    };
-
     ctx->client = NULL;
     size_t errstr_len = strlen(errstr);
     h2o_buffer_reserve(&ctx->resp.after_closed, errstr_len);
     memcpy(ctx->resp.after_closed->bytes + ctx->resp.after_closed->size, errstr, errstr_len);
     ctx->resp.after_closed->size += errstr_len;
     ctx->resp.has_content = 1;
 
+    h2o_header_t headers_sorted[] = {
+        {&H2O_TOKEN_CLIENT_WARNING->buf, NULL, h2o_iovec_init(errstr, errstr_len)},
+        {&H2O_TOKEN_CONTENT_TYPE->buf, NULL, {H2O_STRLIT("text/plain; charset=utf-8")}},
```

Comment:
I would appreciate it if you could use `h2o_iovec_init` to construct the value of the header field; i.e. `h2o_iovec_init(H2O_STRLIT(...))`.

The reason is that generally speaking we would like  to refrain from using struct initialization for `h2o_iovec_t`. In the past, we have had code like `h2o_iovec_t v = {"abc"}` that led to bugs.